pkg/services/ngalert/schedule/schedule.go

```diff
@@ -640,8 +413,10 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 		processedStates := sch.stateManager.ProcessEvalResults(ctx, e.scheduledAt, r, results)
 		sch.saveAlertStates(ctx, processedStates)
 		alerts := FromAlertStateToPostableAlerts(processedStates, sch.stateManager, sch.appURL)
-
-		notify(alerts, logger)
+		err = sch.notifier.Send(key, alerts)
+		if err != nil {
+			logger.Error("failed to send alerts to notifier", "err", err)
```

Comment:
Please see my comment above, I think we need a better term that's not sender with the scheduler package otherwise we have the same problem we had with the dispatcher. To me, router *everywhere* is a good compromise but if you don't like that I'd like to see another proposal that's not *sender*.